libs/utils/benches/benchmarks.rs

```diff
@@ -16,5 +29,29 @@ pub fn bench_id_stringify(c: &mut Criterion) {
     });
 }
 
-criterion_group!(benches, bench_id_stringify);
-criterion_main!(benches);
+pub fn bench_warn_slow(c: &mut Criterion) {
+    for enabled in [false, true] {
+        c.bench_function(&format!("warn_slow/enabled={enabled}"), |b| {
+            run_bench(b, enabled).unwrap()
+        });
+    }
+
+    // The actual benchmark.
+    fn run_bench(b: &mut Bencher, enabled: bool) -> anyhow::Result<()> {
+        const THRESHOLD: Duration = Duration::from_secs(1);
+
+        let runtime = tokio::runtime::Builder::new_current_thread() // single is fine, sync IO only
+            .enable_all()
+            .build()?;
+
+        // Test both with and without warn_slow, since we're essentially measuring Tokio scheduling
+        // performance too.
+        if enabled {
+            b.iter(|| runtime.block_on(warn_slow("ready", THRESHOLD, std::future::ready(()))));
```

Comment:
A more realistic benchmark would be a future that completes below the threshold but not instantly.
I wouldn't be surprised if there's some speculative perf optimisations for ready futures.

Reply:
Good catch, this ended up not actually polling the inner sleep future, which has a cost. Adds 88 ns when we actually yield, up from 46 ns. Updated the benchmark in the PR description.